### DISCUSSION TITLE
Update EssentialsX compatible MC versions

### DISCUSSION THREAD

**Discussion by Chris6ix** [EssentialsX/Essentials#4909] [Ideas and Feature Suggestions] [answered]
I looked a bit through the EssX project and saw this in the latest changelog:

![image](https://user-images.githubusercontent.com/65610536/169154951-672ea964-f33f-4c49-98e3-33d12e5e49e4.png)

I was a bit "confused" by the fact that EssX still supports some very old MC versions and asked myself if some of these are even played by anyone, so I looked at the [EssX bStats page](https://bstats.org/plugin/bukkit/EssentialsX/858) and saw that :

5.2% are still playing 1.8.8 (for some weird reason)
7.5% are still playing 1.12.2 (for some weird reason)
0.8% are on 1.15.2
10.9% are on 1.16.(4/5)
11% are on 1.17(.1)
62.4% are on 1.18+
And the other versions are basically not played at all

So I asked myself if the MC versions EssX supports should be changed to be more up to date (and to make more sense)
imo it should look more like this:

✅ `1.18.2` - EssentialsX actively develops against and supports this version.
⚠️ `1.16.5`, and `1.17.1` - these versions are still supported, but are not a priority for us, and may be dropped in a future release.

I'm not a coder, so please correct me if I'm wrong, but wouldn't dropping these ancient MC versions (That barely anyone plays) make EssX easier to maintain and simplify/optimize its code? (At least a dev that dropped old MC version support on a plugin told me that)
Changing it to what I proposed would still include 85% of the playerbase while *maybe/potentially* "modernizing" and simplifying EssX...so please tell me what you guys think of this :)

**Comment by pop4959** [comment]
You're correct, dropping these versions would make maintaining EssentialsX easier, and that is exactly why they are labeled as subject to being dropped at a future time.

These versions are currently mostly supported out of courtesy to the player base still using 1.8 and 1.12. This used to be a much larger number, but as you've shown it has shrunk a lot over the past few years.

Supporting versions in-between is generally not a pain point. Ironically the more difficult part is supporting prior to 1.13 (the "flattening") which includes both 1.8 and 1.12. There is also difficulty in supporting prior to 1.8 due to the non-existence of UUIDs, which is why 1.8 is the earliest version supported by most plugins (or sometimes 1.7.10, when they were added).

**Comment by mdcfe** [comment, marked as answer]
We guarantee support for the last two Minecraft releases at the time of each release. This is because Spigot and Paper tend to only support up to two versions at a time and typically do not update versions older than this.

EssentialsX's compatibility code for 1.8.8-1.17.1 is stable. As it stands, removing this code would take more effort than keeping it around. There isn't really any benefit to telling users on older MC versions that EssentialsX doesn't support those versions - we would still prefer people to use the latest version of EssentialsX so that we can provide help on GitHub and Discord and so that people can easily find the right builds for older servers.

We have soft-dropped support for certain MC versions once before (removing them from the supported versions list despite no known breakages), but in that it was necessary to make further changes for modern MC versions. In the case of 1.8.8-1.16.5, there isn't really a tangible benefit to anyone for dropping these older versions *yet*. I would consider dropping older versions if we wanted to use modern Java features, but so far there hasn't been any need to consider this.

**Reply by Chris6ix** [reply]
Okay, I understand :)
Thank you for this detailed explanation!
